### PR TITLE
TEST-37 SnapshotMethodRule support + Feature installation via SSH

### PR DESCRIPTION
**Pull request by paouelle**
### Requirements
* Filling out the template is required. Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change
Introduced SnapshotMethodRule that extends on JUnit's MethodRule to separate the snapshot from the apply..

Enhanced MethodRuleChain and MethodRuleAnnotationProcessor to handle method rules that are SnapshotMethodRule. Modify ConfigurationAdmin, ServiceAdmin, and InjectedService to implement the new SnapshotMethodRule.

Added support in Dominion for installing features after the container has been started as opposed to installing them as boot features.

### Verification Process

- Successful build

### Applicable Issues

Fixes: #37

### CODE REVIEW

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

File: junit-extensions/src/main/java/org/codice/junit/ClearInterruptions.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 /**
  * The <code>ClearInterruptions</code> annotation can be used in conjunction with the {@link
- * org.codice.junit.MethodRuleAnnotationRunner} JUnit test runner or the {@link
+ * MethodRuleAnnotationRunner} JUnit or Dominion test runner or with the {@link
```
Comment (AzGoalie) [inline junit-extensions/src/main/java/org/codice/junit/ClearInterruptions.java]: ✏️ Nit: Should there be a comma after Dominion test runner like in the next change for consistency?

File: junit-extensions/src/main/java/org/codice/junit/RestoreSystemProperties.java [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 /**
  * The <code>RestoreSystemProperties</code> annotation can be used in conjunction with the {@link
- * org.codice.junit.MethodRuleAnnotationRunner} JUnit test runner or the {@link
+ * MethodRuleAnnotationRunner} JUnit or Dominion test runner or with the {@link
```
Comment (AzGoalie) [inline junit-extensions/src/main/java/org/codice/junit/RestoreSystemProperties.java]: ✏️ Nit: Same as above: comma?

File: junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  /**
+   * Gets the name of the test method being executed.
+   *
+   * @return the name of the test method being executed
+   */
+  public String getName() {
+    final FrameworkMethod m = method;
+
+    return (m != null) ? m.getName() : null;
```
Comment (AzGoalie) [inline junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java]: ❓ Would an empty string be better than null?

File: junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+   *
+   * @return the source file where the test method is defined or <code>null</code> if not available
+   */
+  public String getSourceFile() {
+    final CtClass c = ctclass;
+
+    if (c != null) {
+      return c.getClassFile2().getSourceFile();
+    }
+    return null;
```
Comment (AzGoalie) [inline junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java]: ❓ Same as above: would an empty string be better than null?

File: junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java (hunk 3 above)
Comment (paouelle) [reply junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java]: Not sure I want to return "". I would prefer to leave it to the caller to know what to do with it if unavailable.

File: junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java (hunk 4 above)
Comment (paouelle) [reply junit-extensions/src/main/java/org/codice/junit/rules/AbstractMethodRule.java]: Not sure I want to return "". I would prefer to leave it to the caller to know what to do with it if unavailable.

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now
internal build download issue

**Pull request merged**